src/cysignals/macros.h

```diff
@@ -244,11 +256,15 @@ static inline void sig_unblock(void)
     }
 #endif
     --cysigs.block_sigint;
-
+#if defined(__MINGW32__) || defined(_WIN32)
+    if (unlikely(cysigs.interrupt_received) && cysigs.sig_on_count > 0)
+      raise(cysigs.interrupt_received);  /* Re-raise the signal */
+#else
```

Comment:
Not to call out this one in particular, but I feel like a lot of these places where there are `#ifdef`s would be better replaced with macros of some appropriate name, and the macros moved to implementation-specific header files.  Going to need to think about this more though on the specifics, but some refactoring is definitely in order here or it's going to become unmaintainable.

Reply:
Ok. It looks like you have the same general idea as jeroen "major comment". Advices on the best way to refactor this are welcome.